### PR TITLE
Migrated Newman from argparse to commander for CLI argument parsing.

### PR DESCRIPTION
**Pull request by deepakpathania**
- Moved from argparse to commander for CLI argument parsing.
- Removed legacy options and their tests which were deprecated.
- Moved the entire CLI parsing logic to `bin/newman.js` and deleted `lib/cli/index.js`
- Added and modified tests for commander specific functionality.

### CODE REVIEW

**Comment by deepakpathania** [comment]
Integration tests are failing which are unrelated to the changes made.

**Comment by kamalaknn** [comment]
@deepakpathania the fix for this is merged on upstream. Updating your branch against upstream should fix the tests.

**Comment by deepakpathania** [comment]
@kamalaknn Hey, thanks for the update, adding a few more functionalities, would update the branch against upstream once that is done.

File: bin/newman.js [hunk 1]
```diff
@@ -2,11 +2,261 @@
...
+        };
+    },
+
+    /**
+     *  used for collecting global key=value variables supplied through command line
+     *
+     * --global-var "foo=bar" --global-var "alpha=beta"
+     *
+     * @param {String} val - The argument provided to `--global-var`.
+     * @param {String} memo - The array that is populated by key value pairs.
```
Comment (harryi3t) [inline bin/newman.js]: `@param {Array} memo` ?

File: bin/newman.js [hunk 2]
```diff
@@ -2,11 +2,261 @@
...
+     * @returns {{key, value}} - The object representation of the current CLI variable.
+    */
+    collect = function (val, memo) {
+        var arg = val,
+            eqIndex = arg.indexOf('='),
```
Comment (harryi3t) [inline bin/newman.js]: we can avoid changing the datatype of `arg` by
```js
var arg,
   eqIndex = val.indexOf('='),
   arg = ... ;
```

**Comment by deepakpathania** [comment]
@harryi3t Hey, I have made a separate PR #1278 with certain changes. Can you please review that instead? Closing this one.

File: bin/newman.js [hunk 3]
```diff
@@ -2,11 +2,261 @@
...
+    },
+
+    /**
+     *  used for collecting global key=value variables supplied through command line
+     *
+     * --global-var "foo=bar" --global-var "alpha=beta"
+     *
+     * @param {String} val - The argument provided to `--global-var`.
+     * @param {String} memo - The array that is populated by key value pairs.
+     * @returns {{key, value}} - The object representation of the current CLI variable.
```
Comment (harryi3t) [inline bin/newman.js]: This returns an array right not object?

Comment (deepakpathania) [reply bin/newman.js]: Yeah, it should be `[{key,value}]`.

File: bin/newman.js (hunk 1 above)
Comment (deepakpathania) [reply bin/newman.js]: Yes, that entire comment needs changes.

File: bin/newman.js (hunk 2 above)
Comment (deepakpathania) [reply bin/newman.js]: Noted.

**Pull request closed without merging**